zrml/global-disputes/src/benchmarks.rs

```diff
@@ -0,0 +1,172 @@
+            let locked_balance: BalanceOf<T> = 1u128.saturated_into();
+            vote_locks.push((market_id, locked_balance));
+            <Winners<T>>::insert(market_id, winner_info.clone());
+        }
+        <Locks<T>>::insert(voter, LockInfo(vote_locks));
+    }: _(RawOrigin::Signed(caller), voter_lookup)
+
+    add_vote_outcome {
+        // concious decision for using component 0..MaxOwners here
+        // because although we check that is_finished is false,
+        // Winners counts processing time for the decoding of the owners vector.
+        // then if the owner information is not present on Winners, the owner info is present on Outcomes
+        // this happens in the case, that Outcomes is not none at the query time.
+        let w in 0..T::MaxOwners::get();
+
+        let mut owners = Vec::new();
+        for i in 0..w {
+            let owner = account("winners_owner", i, 0);
+            owners.push(owner);
+        }
+        let owners = BoundedVec::try_from(owners).unwrap();
+        let winner_info = WinnerInfo {outcome: OutcomeReport::Scalar(0), vote_sum: 42u128.saturated_into(), is_finished: false, owners};
+
+        let caller: T::AccountId = whitelisted_caller();
+        let market_id: MarketIdOf<T> = 0u128.saturated_into();
+        let outcome = OutcomeReport::Scalar(20);
+        <Winners<T>>::insert(market_id, winner_info);
+        deposit::<T>(&caller);
+    }: _(RawOrigin::Signed(caller.clone()), market_id, outcome)
+
+    reward_outcome_owner {
+        let o in 0..T::MaxOwners::get();
+
+        let k in 0..T::RemoveKeysLimit::get();
+
+        let market_id: MarketIdOf<T> = 0u128.saturated_into();
+
+        for i in 0..=k {
+            let owner = account("outcomes_owner", i, 0);
+            GlobalDisputes::<T>::push_voting_outcome(&market_id, OutcomeReport::Scalar(i.into()), &owner, 1_000u128.saturated_into());
+        }
+
+        let mut owners = Vec::new();
+        for i in 0..o {
+            let owner = account("winners_owner", i, 0);
+            owners.push(owner);
+        }
+        let owners = BoundedVec::try_from(owners).unwrap();
+        let winner_outcome = OutcomeReport::Scalar(0);
+
+        let outcome_info = OutcomeInfo {outcome_sum: 42u128.saturated_into(), owners};
+        <Outcomes<T>>::insert(market_id, winner_outcome.clone(), outcome_info);
+
+        let winner_info = WinnerInfo {outcome: winner_outcome, vote_sum: 42u128.saturated_into(), is_finished: true, owners: Default::default()};
+        <Winners<T>>::insert(market_id, winner_info);
+
+        let reward_account = GlobalDisputes::<T>::reward_account(&market_id);
+        let _ = T::Currency::deposit_creating(&reward_account, 100_000u128.saturated_into());
+
+        let caller: T::AccountId = whitelisted_caller();
+
+        let outcome = OutcomeReport::Scalar(20);
+
+        deposit::<T>(&caller);
+    }: _(RawOrigin::Signed(caller.clone()), market_id)
+}
+
```

Comment:
Suggestion: Verify after your benchmark that the state matches your expectations (see [Testing benchmarks](https://docs.substrate.io/test/benchmark/) section in the Substrate documentation)

Reply:
fdfb4c6e366ee4e27b92b083143ce896dcb03cb7

Reply:
Nice :100: